pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -225,71 +225,30 @@ func (r *ReconcileWindowsMachine) addWorkerNode(ipAddress, instanceID string) er
 
-	// sshd service is started to create the default sshd_config file. This file is modified
-	// for enabling publicKey auth and the service is restarted for the changes to take effect.
-	userDataSecret := &core.Secret{
-		ObjectMeta: meta.ObjectMeta{
-			Name:      "windows-user-data",
-			Namespace: "openshift-machine-api",
-		},
-		Data: map[string][]byte{
-			"userData": []byte(`<powershell>
-			Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
-			$firewallRuleName = "ContainerLogsPort"
-			$containerLogsPort = "10250"
-			New-NetFirewallRule -DisplayName $firewallRuleName -Direction Inbound -Action Allow -Protocol TCP -LocalPort $containerLogsPort -EdgeTraversalPolicy Allow
-			Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
-			Install-Module -Force OpenSSHUtils
-			Set-Service -Name ssh-agent -StartupType ‘Automatic’
-			Set-Service -Name sshd -StartupType ‘Automatic’
-			Start-Service ssh-agent
-			Start-Service sshd
-			$pubKeyConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PubkeyAuthentication yes','PubkeyAuthentication yes'
-			$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
- 			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
-			$passwordConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			$authFileConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys','#AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys'
-			$authFileConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			$pubKeyLocationConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'Match Group administrators','#Match Group administrators'
-			$pubKeyLocationConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			Restart-Service sshd
-			New-item -Path $env:USERPROFILE -Name .ssh -ItemType Directory -force
-			echo "` + string(pubKeyBytes[:]) + `"| Out-File $env:USERPROFILE\.ssh\authorized_keys -Encoding ascii
-			</powershell>
-			<persist>true</persist>`),
-		},
+	if err != nil {
+		return errors.Errorf("could not find windows user data secret in required namespace: %v", err)
 	}
 
-	// check if the userDataSecret already exists
-	err := r.client.Get(context.TODO(), kubeTypes.NamespacedName{Name: userDataSecret.Name, Namespace: userDataSecret.Namespace}, &core.Secret{})
-	if err != nil {
-		if k8sapierrors.IsNotFound(err) {
-			log.Info("Creating a new Secret", "Secret.Namespace", userDataSecret.Namespace, "Secret.Name", userDataSecret.Name)
-			err = r.client.Create(context.TODO(), userDataSecret)
-			if err != nil {
-				return errors.Wrap(err, "error creating windows user data secret")
-			}
-		}
-		return errors.Wrap(err, "error creating windows user data secret")
+	secretData := string(instance.Data["userData"][:])
+	if !strings.Contains(secretData, string(pubKeyBytes[:])) {
```

Comment:
Are you just checking for pubKeyBytes? I think we also need to look at the entire existing secret or is the scope of this controller limited to checking the pulickey?

Reply:
Its not. I will update the check.